src/admin/shared/hoc/with-admin-core-config.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+	AdminConfig,
+	AdminConfigManager,
+	AvoOrHetArchief,
+	CommonUser,
+	ContentBlockType,
+	LinkInfo,
+	ToastInfo,
+} from '@meemoo/admin-core-ui';
+import { Icon, IconName, IconProps, Spinner } from '@viaa/avo2-components';
+import { Avo } from '@viaa/avo2-types';
+import { compact, noop } from 'lodash-es';
+import React, { ComponentType, FunctionComponent, useCallback, useEffect, useState } from 'react';
+import { withRouter } from 'react-router';
+import { Link, useHistory, useParams } from 'react-router-dom';
+import { compose } from 'redux';
+
+import { toAbsoluteUrl } from '../../../authentication/helpers/redirects';
+import { APP_PATH, RouteId } from '../../../constants';
+import { getEnv } from '../../../shared/helpers';
+import { tHtml } from '../../../shared/helpers/translate';
+import withUser from '../../../shared/hocs/withUser';
+import useTranslation from '../../../shared/hooks/use-translation/use-translation';
+import { AssetsService } from '../../../shared/services/assets-service/assets.service';
+import { SmartschoolAnalyticsService } from '../../../shared/services/smartschool-analytics-service';
+import { ToastService } from '../../../shared/services/toast-service';
+import { navigationService } from '../../navigation/services/navigation-service';
+import { PermissionsService } from '../services/permissions';
+import { UserGroupsService } from '../services/user-groups';
+
+const InternalLink = (linkInfo: LinkInfo) => {
+	return <Link {...linkInfo} />;
+};
+
+export const withAdminCoreConfig = (WrappedComponent: ComponentType): ComponentType => {
+	const Component = (props: Record<string, unknown>) => {
+		const [adminCoreConfig, setAdminCoreConfig] = useState<AdminConfig | null>(null);
+		const user = props.user as Avo.User.User | undefined;
```

Comment:
I'd prefer to avoid this cast and make props more strongly typed
```
props: {
  user?: Avo.User.User,
  [key: string]: unknown
}
```
Ideally we can infer the rest of the props from `ComponentType` but that might require a lot of dynamic typing.